pkg/controllers/management/auth/globalroles/globalrolebinding_handler_test.go

```diff
@@ -1146,3 +1175,590 @@ func Test_reconcileClusterPermissions(t *testing.T) {
 	}
 
 }
+
+func Test_reconcileNamespacedPermissions(t *testing.T) {
+	t.Parallel()
+
+	activeNamespace := &corev1.Namespace{
+		Status: corev1.NamespaceStatus{
+			Phase: corev1.NamespaceActive,
+		},
+	}
+	terminatingNamespace := &corev1.Namespace{
+		Status: corev1.NamespaceStatus{
+			Phase: corev1.NamespaceTerminating,
+		},
+	}
+
+	tests := []struct {
```

Comment:
Nit: Would be good to have a test case for removing invalid roles from terminating namespaces (and potentially one to ensure that we still create roles in other namespaces should the opportunity arise).

Reply:
Added